tests/python/pants_test/engine/test_scheduler.py

```diff
@@ -224,15 +240,55 @@ def test_get_type_match_failure(self):
 
   def test_unhashable_failure(self):
     """Test that unhashable Get(...) params result in a structured error."""
-    expected_msg = """TypeError: unhashable type: 'list'"""
-    with assert_execution_error(self, expected_msg):
+    with assert_execution_error(self, "TypeError: unhashable type: 'list'"):
       self.scheduler.product_request(C, [Params(TypeCheckFailWrapper(A()))])
 
+    # Test that the error contains the full traceback from within the CFFI context as well
+    # (mentioning which specific extern method ended up raising the exception).
+    expected_msg = dedent(
+      """\
+      1 Exception raised in CFFI extern methods:
+      Traceback (most recent call last):
+        File LOCATION-INFO, in __hash__
+          return super(DataType, self).__hash__()
```

Comment:
It seems weird that this appears twice; once in the subclass constructor and once in the superclass; that seems like a pretty noisy thing to be asserting on... Is it possible to get rid of one of them?

Reply:
Sure, I guess I just wanted to ensure the full traceback is printed, since the current `assert_execution_error()` contextmanager *happens* to make matching against the entire string easier than matching multiple substrings of it. I think this could and should be converted to just checking the one traceback entry (which can then be the same for both py2 and py3), and then separately that the exception message itself is as we expect.

Reply:
I've modified the testing to assert something much less brittle, and in the process extended the assertions to cover the other ways that method can fail!